ResearchKit/Common/ORKAnswerFormat.h

```diff
@@ -151,8 +151,8 @@ ORK_CLASS_AVAILABLE
 
 + (ORKTextAnswerFormat *)textAnswerFormatWithMaximumLength:(NSInteger)maximumLength;
 
-+ (ORKTextAnswerFormat *)textAnswerFormatWithValidationRegex:(NSString *)validationRegex
-                                              invalidMessage:(NSString *)invalidMessage;
++ (ORKTextAnswerFormat *)textAnswerFormatWithValidationRegularExpression:(NSRegularExpression *)validationRegularExpression
```

Comment:
I agree with @syoung-smallwisdom that deprecation flags helps notify developers of API change, so they can eventually conform to it without having a broken build. However, it also encourages them to continue using deprecated API. The best way to encourage developers to conform to new API is to use the compiler to notify them of the API change via a build failure.

That being said, I'm indifferent about which approach we take here, as they both provide value in different ways.